### PR TITLE
add hw-51-clone

### PR DESCRIPTION
**Pull request by TarasHryts**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+        this.name = name;
+        this.color = color;
+        this.speed = speed;
+        this.type = type;
+        this.engine = engine;
+    }
+
+    @Override
+    public Car clone() throws CloneNotSupportedException {
+        return new Car("Audi", "red", 200, "sport", new Engine());
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Car.java]: we shouldn't initialized car

File: src/main/java/core/basesyntax/Engine.java [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
 
-public class Engine {
+public class Engine implements Cloneable {
+    private String name;
+    private Integer id;
+    private Integer volume;
+    private String type;
+    private Integer numberOfCylinders;
 
+    public Engine() {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Engine.java]: the same

File: src/main/java/core/basesyntax/Car.java (hunk 1 above)
Comment (TarasHryts) [reply src/main/java/core/basesyntax/Car.java]: done

File: src/main/java/core/basesyntax/Engine.java (hunk 2 above)
Comment (TarasHryts) [reply src/main/java/core/basesyntax/Engine.java]: done

File: src/main/java/core/basesyntax/Car.java [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
+        this.name = name;
+        this.color = color;
+        this.speed = speed;
+        this.type = type;
+        this.engine = engine;
+    }
+
+    @Override
+    protected Car clone() throws CloneNotSupportedException {
+        return (Car)super.clone();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Car.java]: what about engine? we should call method clone for it

File: src/main/java/core/basesyntax/Engine.java [hunk 4]
```diff
@@ -1,5 +1,22 @@
...
+        this.name = "hybrid";
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Engine.java]: it is bad practice to initialize fields in the constructor

Comment (TarasHryts) [reply src/main/java/core/basesyntax/Engine.java]: done

File: src/main/java/core/basesyntax/Car.java (hunk 3 above)
Comment (TarasHryts) [reply src/main/java/core/basesyntax/Car.java]: done

File: src/main/java/core/basesyntax/Car.java [hunk 5]
```diff
@@ -1,5 +1,22 @@
...
+        this.name = name;
+        this.color = color;
+        this.speed = speed;
+        this.type = type;
+        this.engine = engine;
+    }
+
+    @Override
+    protected Car clone() throws CloneNotSupportedException {
+        return new Car(this.name, this.color, this.speed, this.type, this.engine);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Car.java]:
```suggestion
        return new Car(this.name, this.color, this.speed, this.type, this.engine.clone());
```

**Pull request closed without merging**